src/query/api/v1/handler/placement/common.go

```diff
@@ -479,3 +481,63 @@ func isStateless(serviceName string) bool {
 	}
 	return false
 }
+
+func deleteAggregatorShardSetIDRelatedKeys(
+	svc handleroptions.ServiceNameAndDefaults,
+	svcOpts handleroptions.ServiceOptions,
+	clusterClient clusterclient.Client,
+	shardSetIDs []uint32,
+) error {
+	if svc.ServiceName != handleroptions.M3AggregatorServiceName {
+		return fmt.Errorf("error deleting aggregator instance keys, bad service: %s",
+			svc.ServiceName)
+	}
+
+	kvOpts := kv.NewOverrideOptions().
+		SetEnvironment(svcOpts.ServiceEnvironment).
+		SetZone(svcOpts.ServiceZone)
+
+	flushTimesMgrOpts := aggregator.NewFlushTimesManagerOptions()
+	electionMgrOpts := aggregator.NewElectionManagerOptions()
+	kvStore, err := clusterClient.Store(kvOpts)
+	if err != nil {
+		return fmt.Errorf("cannot get KV store to delete aggregator keys: %v", err)
+	}
+
+	for _, shardSetID := range shardSetIDs {
+		// Check if flush times key exists, if so delete.
+		flushTimesKey := fmt.Sprintf(flushTimesMgrOpts.FlushTimesKeyFmt(),
+			shardSetID)
+		_, flushTimesKeyErr := kvStore.Get(flushTimesKey)
+		if flushTimesKeyErr != nil && flushTimesKeyErr != kv.ErrNotFound {
+			return fmt.Errorf(
+				"error check flush times key exists for deleted instance: %v",
+				flushTimesKeyErr)
+		}
+		if flushTimesKeyErr == nil {
+			// Need to delete the flush times key.
+			if _, err := kvStore.Delete(flushTimesKey); err != nil {
+				return fmt.Errorf(
+					"error delete flush times key for deleted instance: %v", err)
+			}
+		}
+
+		// Check if election manager lock key exists, if so delete.
+		electionKey := fmt.Sprintf(electionMgrOpts.ElectionKeyFmt(),
+			shardSetID)
+		_, electionKeyErr := kvStore.Get(electionKey)
+		if electionKeyErr != nil && electionKeyErr != kv.ErrNotFound {
+			return fmt.Errorf(
+				"error checking election key exists for deleted instance: %v", err)
```

Comment:
Think this would work better as a multierror, so you can still delete the rest of the election keys if one errors?

Reply:
Agreed, will do.